src/herder/TxSetFrame.cpp

```diff
@@ -33,23 +34,26 @@ namespace stellar
 
 using namespace std;
 
+TxSetFrame::TxSetFrame(Hash const& previousLedgerHash,
+                       Transactions const& transactions)
+    : mPreviousLedgerHash(previousLedgerHash)
+    , mTxsInHashOrder(TxSetUtils::sortTxsInHashOrder(transactions))
+    , mHash(
+          TxSetUtils::computeContentsHash(previousLedgerHash, mTxsInHashOrder))
+{
+}
+
 TxSetFrame::TxSetFrame(Hash const& previousLedgerHash)
-    : mHash(std::nullopt)
-    , mValid(std::nullopt)
-    , mPreviousLedgerHash(previousLedgerHash)
+    : TxSetFrame(previousLedgerHash, Transactions{})
 {
 }
 
 TxSetFrame::TxSetFrame(Hash const& networkID, TransactionSet const& xdrSet)
-    : mHash(std::nullopt), mValid(std::nullopt)
+    : mPreviousLedgerHash(xdrSet.previousLedgerHash)
+    , mTxsInHashOrder(TxSetUtils::sortTxsInHashOrder(networkID, xdrSet))
```

Comment:
You probably need a dedicated function that allows to normalize things: we should really be explicit when we use this in production code.
It's fine if we construct another `TxSetFrame` (instead of "in place" normalization with the old code when calling `sortForHash`)

Reply:
It's not very clear to me what should be done here. 
My understanding here, and from the two "check sorted" discussions [above1](https://github.com/stellar/stellar-core/pull/3395#discussion_r857779243) and [above2](https://github.com/stellar/stellar-core/pull/3395#discussion_r857789597), it is possible we might receive a bad `xdrSet` (out of hash order), in which case, we should not normalize it and make it valid (and altering the set and introducing malleability attack)? I'm not getting the "dedicated function that allows to normalize things" part.

Reply:
Normalization is in the context of tests (maybe also in the context of TransactionQueue), as we construct a `TxSetFrame` from a vector of transactions, and we just want to construct a valid `TxSetFrame` in that case

Reply:
Turns out we didn't really need a dedicated normalization method. The one place where I did such a normalization was done [here](https://github.com/stellar/stellar-core/blob/928947a06a10a8bc44daea96166ebe70c39befd0/src/herder/test/HerderTests.cpp#L336-L353), which is done by transforming `xdrSet` to `vector<TransactionFrameBasePtr>` and call `TxSetFrame` constructor on that.